### DISCUSSION TITLE
Meeing notes #4, 2022-11-14

### DISCUSSION THREAD

**Discussion by jongfeel** [fkdl0048/Mentoring-Project#21] [General]
### 동료의 코드를 보고 해줘야 할 이야기

코드가 너무 단순하고 하드코딩적인 부분은 제외하고라도
update 문에서 component를 가져오는 코드가 이해가 잘 안됨

=>

그 분에게 무엇이 잘못됐는지 정확하게 인지를 시켜주는게 중요
지금 이 마음에 안드는 코드를 수정하는게 좋을지 vs 일단 덮어두고 구현해야 하는 기능에 더 집중해야 하는지 선택

만약 단기 프로젝트라면 잘못되도 일단 두는 방향으로 진행
대신, 잘못됐다는 것도 알고 있어야 하고 다음 부터는 이런 일이 되도록 일어나지 않도록
미리 더 생각을 하고 코드를 작성하고 코드 리뷰를 진행하면서 바로바로 조금씩 수정해서 코드를 작성하는 걸 추천

### 객체지향에 대해서

지금 당장은 상속은 괜찮은 작전임.
만약 거대한 클래스를 만들고 그 클래스를 상속하는 클래스들이 많다는 걸 나중에 느낄 때는 늦은 시기

=>

작은 기능을 하는 단위로 클래스를 잘 나누고
되도록 interface를 잘 사용하자

추가)
class code에 집중하지 말고 class에서 만들어진 객체들의 활동에 집중하는 것이 객체지향의 핵심
객체가 노출한 public method를 통해 객체 자체가 어떤 변화가 일어나고
이 변화가 다른 객체에게 또 어떤 영향을 주도록 메시지(이벤트)를 연결 시켜 놨는지를 생각하고
데이터의 흐름을 주목하는 것이 중요.

나중에 객체지향 관련된 중/고급 책을 한번 읽어볼 것을 추천.

### 기획자와 구현 기능과 일정 타협하기

특정 기능에 대해 이틀 째 구현이 안되고 있는데 어떻게 얘기하는게 좋을지에 대한 방안

=>

하드코딩 혹은 특정 위치에서만 캐릭터가 움직이게 일단 만들어 두고
editor에서 배치해서 구현하는 실제 기능은 나중에 추가해서 타협점을 만드는게 중요

일이 되고 안되고는 같이 일하는 사람들과 협력, 피드백이 중요합니다.

### 객체지향에 대해 추가로 얘기할 부분

우선 아직 경험이 많이 부족하고 알아야 할 것도 많은 시기이기 때문에
많은 걸 알아야 한다는 마음가짐 보다는
경험을 통해 내가 어느정도 알고 있고, 이 수준에서 어떤 책을 봤을 때 이해가 되는 수준인지를 알고 난 후에
부족한 부분이나 모르는 부분을 더 익혀 나가는 게 중요.

객체지향 SOLID 원칙에 대해 너무 심취하지는 말고
이런 원칙이 있고 이런 방향으로 구현하는게 객체지향 원칙의 한 방향이라는 것 정도만 확인

**Comment by fkdl0048** [comment]
감사합니다..!
